packages/build-tools/src/utils/project.ts

```diff
@@ -137,11 +135,36 @@ export async function isUsingYarn2(projectDir: string): Promise<boolean> {
   return (await fs.pathExists(yarnrcPath)) || (await fs.pathExists(yarnrcRootPath));
 }
 
+export function runExpoCliCommand<TJob extends Job>(
+  ctx: BuildContext<TJob>,
+  args: string[],
+  options: SpawnOptions
+): SpawnPromise<SpawnResult> {
+  if (
+    ctx.env.EXPO_USE_LOCAL_CLI !== '0' &&
+    ctx.appConfig.sdkVersion &&
+    semver.satisfies(ctx.appConfig.sdkVersion, '>=46')
+  ) {
+    const argsWithExpo = ['expo', ...args];
+    if (ctx.packageManager === PackageManager.NPM) {
+      return spawn('npx', argsWithExpo, options);
+    } else if (ctx.packageManager === PackageManager.YARN) {
+      return spawn('yarn', argsWithExpo, options);
+    } else if (ctx.packageManager === PackageManager.PNPM) {
+      return spawn('pnpm', ['dlx', ...argsWithExpo], options);
+    } else {
+      throw new Error(`Unsupported package manager: ${ctx.packageManager}`);
+    }
+  } else {
+    return ctx.runGlobalExpoCliCommand(args.join(' '), options);
+  }
+}
+
 async function runExpoDoctor<TJob extends Job>(ctx: BuildContext<TJob>): Promise<SpawnResult> {
   ctx.logger.info('Running "expo doctor"');
   let timeout: NodeJS.Timeout | undefined;
   try {
-    const promise = ctx.runExpoCliCommand('doctor', {
+    const promise = runExpoCliCommand(ctx, ['doctor'], {
```

Comment:
doctor should still use the global cli for now. we will move it to a dedicated cli in the future, then it will be invoked with `npx expo-doctor` most likely.

if you invoke this using the local cli, you will see:

```
╰─$ npx expo doctor

  $ expo doctor is not supported in the local CLI, please use expo-cli doctor instead
```